waku/common/filter.go

```diff
@@ -252,3 +250,8 @@ func IsPubKeyEqual(a, b *ecdsa.PublicKey) bool {
 	// the curve is always the same, just compare the points
 	return a.X.Cmp(b.X) == 0 && a.Y.Cmp(b.Y) == 0
 }
+
+// ValidatePublicKey checks the format of the given public key.
+func ValidatePublicKey(k *ecdsa.PublicKey) bool {
+	return k != nil && k.X != nil && k.Y != nil && k.X.Sign() != 0 && k.Y.Sign() != 0
+}
```

Comment:
I presume you chose to move this here because `filter.go` has the most call to the `ValidatePublicKey()` function. I've spotted a number of functions like this that don't really belong to any one file. I wonder if moving them to a `helper.go` may be a more suitable place for them. What do you think?

Reply:
yes, that makes more sense, I can gather them up and just add to `helper.go`